Fix #557 - public type implements an internal interface that uses an internal type

kurtschelfthout opened the pull request:
I'd like to request a review for this.

The change passes the test I added, but not sure the approach is ok. If you think this is promising, I'd want to add more tests too.

Issue #557

dsyme commented:
@kurtschelfthout Wow, it would be **great** to have this fixed.  The approach looks like the correct one.

kurtschelfthout commented:
(re: edit of title: "There is no try. Do or don't." :) )

So I added more tests and moved them to a more appropriate place (I think). It seems to work without any further compiler changes.

Is the CI still borked? Some things failed last push but I couldn't really find the actual problem or even whether it's something I caused.

In any case this is ready for review from my perspective - I had some trouble coming up with good negative test cases or edge cases where things may go wrong, so feel free to suggest things along those lines (or anything else of course).

KevinRansom commented on tests/fsharpqa/Source/Conformance/DeclarationElements/AccessibilityAnnotations/basic/InterfaceImplementationVisibility.fs:
```diff
@@ -0,0 +1,84 @@
+
+module Definitions =
```
should this be module internal?

kurtschelfthout replied:
That could be an additional test, but for the purpose of this fix I was trying to make things as clear as possible by keeping everything public except the interfaces/types themselves.

KevinRansom commented:
@kurtschelfthout 
Thanks for doing this

Kevin
